edit-post/components/sidebar/post-taxonomies/taxonomy-panel.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * External Dependencies
+ */
+import { connect } from 'react-redux';
+import { get } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { PanelBody } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import { isEditorSidebarPanelOpened } from '../../../store/selectors';
+import { toggleGeneralSidebarEditorPanel } from '../../../store/actions';
+
+function TaxonomyPanel( { taxonomy, isOpened, onTogglePanel, panelName, children } ) {
+	const taxonomyMenuName = get( taxonomy, [ 'labels', 'menu_name' ] );
+	if ( ! taxonomyMenuName ) {
+		return;
+	}
+	return (
+		<PanelBody
+			key={ panelName }
+			title={ taxonomyMenuName }
+			opened={ isOpened }
+			onToggle={ onTogglePanel }
+		>
+			{ children }
+		</PanelBody>
+	);
+}
+
+export default connect(
+	( state, ownProps ) => {
+		const slug = get( ownProps.taxonomy, [ 'slug' ] );
+		const panelName = slug ? `taxonomy-panel-${ slug }` : '';
+		return {
+			panelName,
+			isOpened: slug ? isEditorSidebarPanelOpened( state, panelName ) : false,
+		};
+	},
+	{
+		onTogglePanel( panelName ) {
+			return toggleGeneralSidebarEditorPanel( panelName );
+		},
+	},
+	( stateProps, dispatchProps, ownProps ) => {
```

Comment:
See #5090 and https://github.com/WordPress/gutenberg/pull/5098#discussion_r168808765 for more on my distaste for the `mergeProps` function. Specifically that returning a function here will cause the component to _always_ re-render. Via `withDispatch` of #5137, we can probably avoid this now.